src/functional/program_processors/codegens/gtfn/gtfn_module.py

```diff
@@ -48,25 +52,87 @@ def __call__(
     ) -> stages.ProgramSource[languages.Cpp, languages.LanguageWithHeaderFilesSettings]:
         """Generate GTFN C++ code from the ITIR definition."""
         program = inp.program
-        parameters = tuple(
+        parameters: list[
```

Comment:
Please refactor this function into smaller pieces. There is a lot going on now and I can't easily follow.

Reply:
Done. It's still not a beauty, but should be fine now.

Reply:
I believe if you extract the 2 loops into functions it's as good as it can be with the problematic interface regarding scalar args.